3scale/porta
# Adding autocomplete to params using x-data-threescale-name field

**What this PR does / why we need it**:

Implements autocomplete of API keys, using the `x-data-threescale-name` field (see [docs](https://access.redhat.com/documentation/en-us/red_hat_3scale_api_management/2.8/html/providing_apis_in_the_developer_portal/create-activedocs-spec#activedocs_auto_fill_of_api_keys))
This PR is a continuation of the job started in https://github.com/3scale/porta/pull/1935, later reverted in https://github.com/3scale/porta/pull/1989

**Which issue(s) this PR fixes** 

https://issues.redhat.com/browse/THREESCALE-4857

**Verification steps** 
- Go to a Service's active docs section: apiconfig/services/2/api_docs
- Create a new OAS v3 using the `x-data-threescale-name` fields, as described [here](https://access.redhat.com/documentation/en-us/red_hat_3scale_api_management/2.8/html/providing_apis_in_the_developer_portal/create-activedocs-spec#activedocs_auto_fill_of_api_keys). You can use this [example](https://gist.github.com/damianpm/9c98e6710994162474a5171cd326465b)
- Add a new page or replace the default **Documentation** page with the following snippet (see [docs](https://access.redhat.com/documentation/en-us/red_hat_3scale_api_management/2.8/html/providing_apis_in_the_developer_portal/using-oas30#configuring_the_developer_portal_with_oas_3_0)):
```
{% content_for javascripts %}
    {{ 'active_docs.js' | javascript_include_tag }}
{% endcontent_for %}

<h1>Documentation</h1>

<div class="swagger-section">
  <div id="message-bar" class="swagger-ui-wrap">&nbsp;</div>
  <div id="swagger-ui-container" class="swagger-ui-wrap"></div>
</div>

<script type="text/javascript">
  (function () {
    var url = "{{provider.api_specs.first.url}}";

    SwaggerUI({ url: url, dom_id: "#swagger-ui-container" });
  }());
</script>
```

**Special notes for your reviewer**:

To switch from `user_key` to `app_id` / `app_key`, authentication, go to the edit integration settings page (/apiconfig/services/2/integration) and select an authentication method:
![edit-settings](https://user-images.githubusercontent.com/13486237/85729442-1db90c80-b6f9-11ea-84ac-4cd5703c16ba.png)
____
**Admin Portal:**

![admin-user-key](https://user-images.githubusercontent.com/13486237/85729587-3cb79e80-b6f9-11ea-9ee2-7fb110c305ef.png)
____
![admin-app-id](https://user-images.githubusercontent.com/13486237/85729644-47723380-b6f9-11ea-9f34-ef5eca2ad0b6.png)
____

**Developer Portal:**

![developer-user-key](https://user-images.githubusercontent.com/13486237/85729736-5822a980-b6f9-11ea-8991-85638b211e1e.png)
____
![developer-app-id](https://user-images.githubusercontent.com/13486237/85729759-5ce75d80-b6f9-11ea-97f6-e536abaf89ea.png)
